src/hls-adapter.js

```diff
@@ -925,4 +929,19 @@ export default class HlsAdapter extends BaseMediaSourceAdapter {
       return 0;
     }
   }
+
+  /**
+   * called when a fragment is loaded
+   * @private
+   * @param {any} data - the event data of the loaded fragment
+   * @returns {void}
+   */
+  _onFragLoaded(data): void {
+    const fragmentDownloadTime = data.stats.tload - data.stats.trequest;
+    this._trigger(EventType.FRAG_LOADED, {miliSeconds: fragmentDownloadTime, bytes: data.stats.loaded});
+  }
+
+  get targetBuffer(): number {
+    return Math.min(this._hls.config.maxMaxBufferLength, this._videoElement.duration - this._videoElement.currentTime);
```

Comment:
this is correct when taking into account the edge case where in VOD the playback reaches the time where `duration-currentTime` < `maxMaxBufferLength`.
But the above is actually the stats logic, as the `maxMaxBufferLength` is static, even if the actual buffer size left is smaller the `maxMaxBufferLength`, so I suggest moving this to the stats manager (to-be) in playkit.

Now, for live there's a different story as hls.js has other config for live buffer, so we need to take it from one of the `hls.config.*live*` configs, and we already do it in `getStartTimeOfDvrWindow`.
Move the `maxLatency` calculation to a standalone private method, use it in `getStartTimeOfDvrWindow` and also in this method.
If we are playing live and not in DVR then take `maxLatency` otherwise take `maxMaxBufferLength`